tokenclient/tokenFlows.go

```diff
@@ -31,6 +33,21 @@ type TokenFlows struct {
 	identity env.Identity
 	options  Options
 	tokenURI string
+	cache    *cache.Cache
+}
+
+type request struct {
+	targetURL   string
+	parameters  string
+	httpRequest *http.Request
```

Comment:
It seems like we would be storing some info twice.
Would it be possible to remove targetUrl and parameters and use the values from `http.Request` instead? 
This would allow us to simplify things a bit, e.g. to remove our custom `newRequestWithContext` function.

How about promoting http.Request so we can omit one indirection:
```suggestion
	http.Request
```

Reply:
To be more specific: could we replace `targetURL` with `http.Request.URL` and `parameters` with `http.Request.GetBody()`?

Reply:
🥇 you found my hack :grinning: 
Was able to fix it according to your suggestion! 
Btw: `http.Request.Body` leads to `net/http: make "http: ContentLength=N with Body length 0"` error when performing the request - it seems that the reader is already closed and multiple reads are not possible. But `http.Request.GetBody()` works properly. 👍